### PR TITLE
Add WAVM support.

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

File: bazel/envoy_select.bzl [hunk 1]
```diff
@@ -32,6 +32,21 @@ def envoy_select_hot_restart(xs, repository = ""):
         "//conditions:default": xs,
     })
 
+# Selects the given values depending on the WASM runtimes enabled in the current build.
+def envoy_select_wasm_wavm(xs):
```
Comment (PiotrSikora) [inline bazel/envoy_select.bzl:36]: You could just use `envoy_select_wasm_wavm_or(xs, [])` instead.

Comment (jplevyak) [reply bazel/envoy_select.bzl]: done

File: test/extensions/filters/http/wasm/test_data/BUILD [hunk 2]
```diff
@@ -38,3 +38,11 @@ wasm_cc_binary(
         "@proxy_wasm_cpp_sdk//contrib:contrib_lib",
     ],
 )
+
+genrule(
```
Comment (jplevyak) [inline test/extensions/filters/http/wasm/test_data/BUILD:42]: Could I get some bazel help here?

Comment (PiotrSikora) [reply test/extensions/filters/http/wasm/test_data/BUILD:42]: `//bazel/foreign_cc:wavm` builds only the library, you'd need to add `cc_binary` target for the tool if you want to build it.

Comment (jplevyak) [reply test/extensions/filters/http/wasm/test_data/BUILD:42]: it does build the binary (I checked).  I have been messing with this for a few hours and bazel is just not my thing.  I'm going to rip this out and we can add it in later.

File: bazel/repositories.bzl [hunk 3]
```diff
@@ -824,12 +827,21 @@ def _upb():
...
 def _proxy_wasm_cpp_host():
     _repository_impl(
         name = "proxy_wasm_cpp_host",
         build_file = "@envoy//bazel/external:proxy_wasm_cpp_host.BUILD",
     )
+    #native.new_local_repository(
+    #    name = "proxy_wasm_cpp_host",
+    #    build_file = "@envoy//bazel/external:proxy_wasm_cpp_host.BUILD",
+    #    path = "/home/jplevyak/work/proxy-wasm-cpp-host",
+    #)
```
Comment (PiotrSikora) [inline bazel/repositories.bzl:844]: Debug leftover?

File: bazel/repositories.bzl [hunk 4]
```diff
@@ -824,12 +827,21 @@ def _upb():
 
 def _proxy_wasm_cpp_sdk():
     _repository_impl(name = "proxy_wasm_cpp_sdk")
+    #native.local_repository(
+    #    name = "proxy_wasm_cpp_sdk",
+    #    path = "/home/jplevyak/work/proxy-wasm-cpp-sdk",
+    #)
```
Comment (PiotrSikora) [inline bazel/repositories.bzl:833]: Debug leftover?

File: bazel/repositories.bzl (hunk 3 above)
Comment (jplevyak) [reply bazel/repositories.bzl]: done

File: bazel/repositories.bzl (hunk 4 above)
Comment (jplevyak) [reply bazel/repositories.bzl]: done

**Pull request merged**